src/main/java/net/consensys/orion/cmd/Orion.java

```diff
@@ -431,44 +431,9 @@ public void run(final PrintStream out, final PrintStream err, final Config confi
       options.setPemKeyCertOptions(pemKeyCertOptions);
 
       if (!config.tlsServerChain().isEmpty()) {
-        final PemTrustOptions pemTrustOptions = new PemTrustOptions();
-        for (final Path chainCert : config.tlsServerChain()) {
-          pemTrustOptions.addCertPath(chainCert.toAbsolutePath().toString());
-        }
-        options.setPemTrustOptions(pemTrustOptions);
-      }
-
-      final Path knownClientsFile = config.tlsKnownClients();
-      final String serverTrustMode = config.tlsServerTrust().toLowerCase();
-      switch (serverTrustMode) {
-        case "whitelist":
-          options.setTrustOptions(VertxTrustOptions.whitelistClients(knownClientsFile, false));
-          break;
-        case "ca":
-          // use default trust options
-          break;
-        case "ca-or-whitelist":
-          options.setTrustOptions(VertxTrustOptions.whitelistClients(knownClientsFile, true));
-          break;
-        case "tofu":
-        case "insecure-tofa":
-          options.setTrustOptions(VertxTrustOptions.trustClientOnFirstAccess(knownClientsFile, false));
-          break;
-        case "ca-or-tofu":
-        case "insecure-ca-or-tofa":
-          options.setTrustOptions(VertxTrustOptions.trustClientOnFirstAccess(knownClientsFile, true));
-          break;
-        case "insecure-no-validation":
-        case "insecure-record":
-          options.setTrustOptions(VertxTrustOptions.recordClientFingerprints(knownClientsFile, false));
-          break;
-        case "insecure-ca-or-record":
-          options.setTrustOptions(VertxTrustOptions.recordClientFingerprints(knownClientsFile, true));
-          break;
-        default:
-          throw new UnsupportedOperationException(
-              "\"" + serverTrustMode + "\" option for tlsservertrust is not supported");
+        options.setPemTrustOptions(createPemTrustOptions(config.tlsServerChain()));
       }
+      options.setTrustOptions(createTrustOptions(config.tlsServerTrust().toLowerCase(), config.tlsKnownClients()));
```

Comment:
Think I remember from discussion that there was potential issue here if createTrustOptions returned a null